types/AbstractBaseType.php

```diff
@@ -385,6 +386,33 @@ public function compare($column, $comp, $value) {
     }
 
     /**
+     * This function is used to modify an aggregation query to add a filter
+     * for the given column matching the given value. A type should add at
+     * least a filter here but could do additional things like joining more
+     * tables needed to handle more complex filters
+     *
+     * @param QueryBuilder $QB the query so far
+     * @param string $column The column name to use in the SQL
+     * @param string $comp The comparator @see Search::$COMPARATORS
+     * @param string $value this is the user supplied value to compare against
+     * @param string $type the logical operator this filter should use (AND|OR)
+     */
+    public function filter(QueryBuilder $QB, $column, $comp, $value, $type) {
+        $pl = $QB->addValue($value);
+
+        switch($comp) {
+            case '~':
+                $QB->filters()->where($type, "$column LIKE $pl");
+                break;
+            case '!~':
+                $QB->filters()->where($type, "$column NOT LIKE $pl");
+                break;
+            default:
+                $QB->filters()->where($type, "$column $comp $pl");
+        }
+    }
```

Comment:
@micgro42 this is a first idea how the QueryBuilder could be passed on to the type (compare with the above compare function for what we currently have). I'm not sure this is enough info, yet. I guess we would also need some tablenames/aliases (eg. the data and multi tables this type's column is part of).

Also this is currently not very flexible in terms of inheritance. A type would always need to implement the function on its own completely and couldn't inherit any functionality from this function (eg. call parent::filter()).

Any better ideas to do this?